.github/workflows/build-and-test.yml

```diff
@@ -32,7 +32,7 @@ jobs:
         key: ${{ runner.os }}-${{ hashFiles('pirouette.cabal', 'cabal.project') }}]
         restore-keys: ${{ runner.os }}-
 
-    - name: Accessing the result cache
+    - name: Accessing the result artifact
```

Comment:
My bad! I meant that currently we're using the cache to send files from one job to another (and therefore the name of the step was fine as it was), when it looks to me like we should go through the artifacts mechanism instead. It would also allow us to download the results of the tests if we need to debug.

Reply:
Maybe we should just keep it as it is for now. It works, after all.

Reply:
There can be good reasons to split the CI in several jobs. My guess here is that it would allow triggering an upload of the Cabal cache even when tests fail, which can indeed be very valuable.

Reply:
Documented in https://github.com/tweag/pirouette/issues/164.